ops/dataset-copy.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+# a script to copy a dataset from one deployment to another
+
+set -eu
+
+### INPUTS ###
+DATASET_UUID_TO_COPY="00000000-0000-0000-0000-000000000000"
+AUTH_TOKEN="authorization: Bearer ya29"
+
+SRC_URL="https://jade.datarepo-dev.broadinstitute.org"
+DST_URL="https://jade-fb.datarepo-dev.broadinstitute.org/"
+
+### utility functions ###
+token_check() {
+    RESULT=$(curl -s -X GET "${SRC_URL}/api/resources/v1/profiles" -H "accept: application/json" -H "${AUTH_TOKEN}")
+    if [[ ${RESULT} =~ "<title>401 Unauthorized</title>" ]]; then
+        echo "Error: AUTH_TOKEN is invalid or expired"
+        exit 1
+    fi
+}
+
+uuid_lc() {
+    uuidgen | tr '[:upper:]' '[:lower:]'
+}
+
+post_url() {
+    curl -s -X POST "${1}" -H  "accept: application/json" -H  "Content-Type: application/json" -H "${AUTH_TOKEN}" -d "${2}"
+}
+
+get_url() {
+    curl -s -X GET  "${1}" -H  "accept: application/json" -H  "Content-Type: application/json" -H "${AUTH_TOKEN}"
+}
+
+CURRENT_DATE=$(date -I)
+CURRENT_DATE=${CURRENT_DATE//-}
+
+### PART 1: create profile ###
+profile_create() {
+    echo "1: creating new profile"
+
+    PROFILE_URL="${DST_URL}/api/resources/v1/profiles"
+
+    PROFILE_UUID=$(uuid_lc)
+    PROFILE_NAME="profile-search-api-${CURRENT_DATE}-${PROFILE_UUID:0:8}"
+
+    PROFILE_JSON=$(cat << EOF
+{
+  "biller": "direct",
+  "billingAccountId": "00708C-45D19D-27AAFA",
+  "description": "profile created by datasetcopy.sh",
+  "id": "${PROFILE_UUID}",
+  "profileName": "${PROFILE_NAME}"
+}
+EOF
+)
+
+    post_url "${PROFILE_URL}" "${PROFILE_JSON}"
+
+    printf "\n1: created profile %s\n" "${PROFILE_UUID}"
+}
+
+### PART 2: copy dataset schema ###
+dataset_copy_schema() {
+    echo "2: copying dataset schema"
+
+    DATASET_URL="${SRC_URL}/api/repository/v1/datasets/${DATASET_UUID_TO_COPY}?include=SCHEMA&include=ACCESS_INFORMATION"
+
+    DATASET_JSON=$(get_url "${DATASET_URL}")
+    DATASET_NAME=$(echo "${DATASET_JSON}" | jq -r '.name')
+    DATASET_SCHEMA=$(echo "${DATASET_JSON}" \
+        | jq 'del(.id, .dataProject, .defaultSnapshotId, .createdDate, .storage, .accessInformation) | . + { "cloudPlatform": "gcp", "region": "US" } | .defaultProfileId = "'"${PROFILE_UUID}"'"' \
+        | sed 's/"datatype": "fileref"/"datatype": "string"/')
+    DATASET_TABLES=$(echo "${DATASET_JSON}" | jq -r '.accessInformation.bigQuery.tables[].qualifiedName')
+    
+    echo "${DATASET_SCHEMA}" > schema.json
+
+    echo "2: copied schema for dataset ${DATASET_UUID_TO_COPY}"
+}
+
+### part 3: extract dataset data ###
+dataset_extract_data() {
+    echo "3: saving dataset into temporary directory"
+
+    DATASET_TMP_DIR=$(mktemp -d)
+    for DATASET_TABLE in ${DATASET_TABLES} ; do
+        DATASET_TMP_JSON0="${DATASET_TMP_DIR}/${DATASET_TABLE}.json"
+        DATASET_TMP_JSONL="${DATASET_TMP_DIR}/${DATASET_TABLE}.jsonl"
+        bq query --nouse_legacy_sql --format=json 'SELECT * FROM `'"${DATASET_TABLE}"'`' > "${DATASET_TMP_JSON0}"
```

Comment:
I'll investigate how to use `bq export`, but I agree it's probably not worth doing now. At a minimum it would require to reorder the creation of the bucket, and I'd need to thoroughly test it.